Fix docker logs hangs after stop container when using journald

Vanient opened the pull request:
Fix when using journald as log-driver, docker logs -f <containerID> hangs
after the container is exited. LogWatcher.WatchProducerGone() returns a
channel receiver to receive notification once container is gone.

fixes https://github.com/docker/for-linux/issues/575

Signed-off-by: Danni Xia <xiadanni1@huawei.com>

**- What I did**
fix https://github.com/docker/for-linux/issues/575

**- How I did it**
Add logWatcher.WatchProducerGone() to receive notification when container is gone.

**- How to verify it**
`docker logs -f $(docker run -d --rm --log-driver journald busybox sh -c "for i in \$(seq 1 10); do echo logline \$i; sleep 1; done")`
Verify that docker logs process can exit.

kolyshkin commented:
@Vanient journald log driver has lots of issues, some (including docker/for-linux#575) are addressed in https://github.com/moby/moby/pull/38859, you are more than welcome to give it a review and test.

kolyshkin commented:
In particular, the fix for docker/for-linux#575 is here: https://github.com/moby/moby/pull/38859/commits/cd13be800910c0e0a2b22d4ebf0240af01097a24

Vanient committed:
Fix docker logs hangs after stop container when using journald

Fix when using journald as log-driver, docker logs -f <containerID> hangs
after the container is exited. LogWatcher.WatchProducerGone() returns a
channel receiver to receive notification once container is gone.

fixes https://github.com/docker/for-linux/issues/575

Signed-off-by: Danni Xia <xiadanni1@huawei.com>

thaJeztah commented:
@kolyshkin I see this was updated after your last comment; PTAL
